### PR TITLE
test: added utils tests

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: tests/unit/layers/common/utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      expect(typeof convertedYaml).toBe('string');
+      expect(safeDumpStub).toHaveBeenCalledTimes(1);
+    });
+  });
+  describe('#replaceYamlFileContent', () => {
+    it('should replace file content with the requested yaml content', function () {
+      // mock
+      const mockYamlFile = 'tests/unit/mock/mockContent.yaml';
+      const mockConvertedJson: IMapProxyJsonDocument = utils.convertYamlToJson(mockYamlFile);
+      const mockLayer: IMapProxyLayer = {
```
Comment (asafMasa) [inline tests/unit/layers/common/utils.spec.ts:58]: Assert (mock layer doesn't exist)

File: tests/unit/layers/common/utils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        name: 'mockLayer',
+        sources: ['mockSource'],
+      };
+      mockConvertedJson.layers.push(mockLayer);
+      // action
+      const convertedYaml: string = utils.convertJsonToYaml(mockConvertedJson);
+      const newConvertedJson: IMapProxyJsonDocument = jsyaml.safeLoad(convertedYaml) as IMapProxyJsonDocument;
+      replaceYamlFileContentStub.mockReturnValueOnce(undefined);
+      const action = () => utils.replaceYamlFileContent(mockYamlFile, convertedYaml);
```
Comment (asafMasa) [inline tests/unit/layers/common/utils.spec.ts:67]: after replace use the method to load and assert mock exists

File: tests/unit/layers/common/utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      // expectation
+      expect(action).not.toThrow();
+      expect(typeof convertedYaml).toBe('string');
+      expect(newConvertedJson.layers).toContainEqual(mockLayer);
+    });
+  });
+  describe('#sortArrayByZIndex', () => {
+    test('should sort an array in numerical order', function () {
+      // mock
+      const layers: IReorderMosaicLayerObject[] = [
```
Comment (asafMasa) [inline tests/unit/layers/common/utils.spec.ts:77]: assert different z order before your change

**Review by asafMasa — CHANGES_REQUESTED** [review]
good job 👍

File: tests/unit/layers/common/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    it('should convert yaml content to json object', function () {
+      // mock
+      const mockYamlFile = 'tests/unit/mock/mockContent.yaml';
+      // action
+      const action = () => utils.convertYamlToJson(mockYamlFile);
+      // expectation
+      expect(typeof action()).toBe('object');
+      expect(safeLoadStub).toHaveBeenCalledTimes(1);
+    });
+    it('should reject with file not found or not exists', function () {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply tests/unit/layers/common/utils.spec.ts:31]: fixed in 50c0b88

File: tests/unit/layers/common/utils.spec.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply tests/unit/layers/common/utils.spec.ts:87]: fixed in d0ecb1d

File: tests/unit/layers/common/utils.spec.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply tests/unit/layers/common/utils.spec.ts:67]: fixed in 5d9d905

File: tests/unit/layers/common/utils.spec.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply tests/unit/layers/common/utils.spec.ts:77]: cannot use the method because the method reads a yaml *file and 'safeLoad' read yaml content

**Review by asafMasa — APPROVED** [review]

**Pull request merged**